packages/worker/src/worker.ts

```diff
@@ -140,6 +140,33 @@ export type ActivityTaskWithContext = ContextAware<{
 
 type CompiledWorkerOptionsWithBuildId = CompiledWorkerOptions & { buildId: string };
 
+/**
+ * Combined error information for {@link Worker.runUntil}
+ */
+export interface CombinedWorkerRunErrorCause {
+  /**
+   * Error thrown by a Worker
+   */
+  workerError: unknown;
+  /**
+   * Error thrown by the wrapped promise or function
+   */
+  innerError: unknown;
+}
+
+/**
+ * Error thrown by {@link Worker.runUntil}
+ */
+export class CombinedWorkerRunError extends Error {
+  public readonly name = 'CombinedWorkerRunError';
+  public readonly cause: CombinedWorkerRunErrorCause;
+
+  constructor(message: string, { cause }: { cause: CombinedWorkerRunErrorCause }) {
+    super(message);
+    this.cause = cause;
+  }
```

Comment:
It might be useful to customize the `toString()` output on this error to include details about both exceptions. I doubt most devs will actually expect and properly decode the CombinedWorkerRunError exception.

Reply:
Node 16.9 and above will auto format cause so that's not needed.